<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.14"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Bactome: copads.genetic.Chromosome Class Reference</title>
<link href="../../tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../jquery.js"></script>
<script type="text/javascript" src="../../dynsections.js"></script>
<link href="../../navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../resize.js"></script>
<script type="text/javascript" src="../../navtreedata.js"></script>
<script type="text/javascript" src="../../navtree.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
  $(document).ready(initResizable);
/* @license-end */</script>
<link href="../../search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../search/searchdata.js"></script>
<script type="text/javascript" src="../../search/search.js"></script>
<link href="../../doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Bactome
   </div>
   <div id="projectbrief">Toolbox for Studying Bacteria, Archaea, and Complex Taxons</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.14 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "../../search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="../../menudata.js"></script>
<script type="text/javascript" src="../../menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('../../',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('df/d45/classcopads_1_1genetic_1_1_chromosome.html','../../');});
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="../../d1/d9f/classcopads_1_1genetic_1_1_chromosome-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">copads.genetic.Chromosome Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for copads.genetic.Chromosome:</div>
<div class="dyncontent">
 <div class="center">
  <img src="../../df/d45/classcopads_1_1genetic_1_1_chromosome.png" usemap="#copads.genetic.Chromosome_map" alt=""/>
  <map id="copads.genetic.Chromosome_map" name="copads.genetic.Chromosome_map">
</map>
 </div></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a84b6b699de5a80fff463cb60386c6d10"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a84b6b699de5a80fff463cb60386c6d10">__init__</a> (self, <a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a03e4ebef1842713f8e051fde835c377e">sequence</a>, <a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#aa88b94f11b83511072d8f79489c6db84">base</a>, <a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a8ffcfbae7a2bee146a0fa43e1889e36a">background_mutation</a>=0.0001)</td></tr>
<tr class="separator:a84b6b699de5a80fff463cb60386c6d10"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab7681f64bdd10f0360ddebd64c001e2c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#ab7681f64bdd10f0360ddebd64c001e2c">rmutate</a> (self, type='point', rate=0.01, start=0, end=-1)</td></tr>
<tr class="separator:ab7681f64bdd10f0360ddebd64c001e2c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2f0fa5fa582d63a29889acbf47511c5"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#aa2f0fa5fa582d63a29889acbf47511c5">kmutate</a> (self, type='point', start=0, end=0, <a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a03e4ebef1842713f8e051fde835c377e">sequence</a>=None, tpos=0)</td></tr>
<tr class="separator:aa2f0fa5fa582d63a29889acbf47511c5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0f30e69054acbb13ce856f23f8785376"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a0f30e69054acbb13ce856f23f8785376">replicate</a> (self)</td></tr>
<tr class="separator:a0f30e69054acbb13ce856f23f8785376"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:a03e4ebef1842713f8e051fde835c377e"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a03e4ebef1842713f8e051fde835c377e">sequence</a></td></tr>
<tr class="separator:a03e4ebef1842713f8e051fde835c377e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa88b94f11b83511072d8f79489c6db84"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#aa88b94f11b83511072d8f79489c6db84">base</a></td></tr>
<tr class="separator:aa88b94f11b83511072d8f79489c6db84"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8ffcfbae7a2bee146a0fa43e1889e36a"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html#a8ffcfbae7a2bee146a0fa43e1889e36a">background_mutation</a></td></tr>
<tr class="separator:a8ffcfbae7a2bee146a0fa43e1889e36a"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><pre class="fragment">Representation of a linear chromosome.

@see: Lim, JZR, Aw, ZQ, Goh, DJW, How, JA, Low, SXZ, Loo, BZL,
Ling, MHT. 2010. A genetic algorithm framework grounded in biology.
The Python Papers Source Codes 2: 6.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00013">13</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a id="a84b6b699de5a80fff463cb60386c6d10"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a84b6b699de5a80fff463cb60386c6d10">&#9670;&nbsp;</a></span>__init__()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def copads.genetic.Chromosome.__init__ </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>sequence</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>base</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>background_mutation</em> = <code>0.0001</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Sets up a chromosome.

@param sequence: a subscriptable object (list or string) representing
    the sequence of the chromosome.
@param base: a subscriptable object (list or string) representing
    allowable entities in the sequence.
@param background_mutation: background mutation rate represented as the 
    probability of number of mutations per base. Default = 0.0001 
    (0.01%).
</pre> 
<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00022">22</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a id="aa2f0fa5fa582d63a29889acbf47511c5"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aa2f0fa5fa582d63a29889acbf47511c5">&#9670;&nbsp;</a></span>kmutate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def copads.genetic.Chromosome.kmutate </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>type</em> = <code>'point'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>start</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>end</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>sequence</em> = <code>None</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>tpos</em> = <code>0</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Known Mutation operator - to simulate a known point, insertion, 
deletion, inversion, gene translocation or gene duplication event.

The required parameters will be determined by the type of mutation:
    - point requires 
- start (the position of the base to mutate)
- sequence (the new base)
    - delete requires
- start (the position of the base to delete)
    - insert requires
- start (the position of the base to begin insertion)
- sequence (list of sequence or a base to insert)
    - invert requires
- start (the position of the base to start inversion)
- end (the position of the base to end inversion)
    - duplicate requires
- start (the position of the starting base to duplicate)
- end (the position of the last base to duplicate)
    - translocate requires
- start (the position of the base to start translocation)
- end (the position of the base to end translocation)
- tpos (the position of the base insert the translocated 
    sequence)

@param type: type of mutation. Accepts 'point' (point mutation), 
    'insert' (insert a base), 'delete' (delete a base), 'invert'
    (invert a stretch of the chromosome), 'duplicate' (duplicate a
    stretch of the chromosome), 'translocate' (translocate a stretch of
    chromosome to another random position). Default = point.
@param start: starting base on the chromosome for mutation.
    Default = 0, start of the genome.
@param end: last base on the chromosome for mutation. Default = 0.
@param sequence: list of bases to change to (point mutation) or to 
    insert (insertion mutation)
@param tpos: position of the chromosome to insert the translocated
    sequence.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00094">94</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<a id="a0f30e69054acbb13ce856f23f8785376"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a0f30e69054acbb13ce856f23f8785376">&#9670;&nbsp;</a></span>replicate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def copads.genetic.Chromosome.replicate </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Replicates (deep copy) the chromosome.

@return: a copy of current chromosome.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00153">153</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<a id="ab7681f64bdd10f0360ddebd64c001e2c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab7681f64bdd10f0360ddebd64c001e2c">&#9670;&nbsp;</a></span>rmutate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def copads.genetic.Chromosome.rmutate </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>type</em> = <code>'point'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>rate</em> = <code>0.01</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>start</em> = <code>0</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>end</em> = <code>-1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Random Mutation operator - to simulate random point, insertion, 
deletion, inversion, gene translocation and gene duplication events.

The start and end parameters are useful for simulating mutational 
hotspots in the genome.

@param type: type of mutation. Accepts 'point' (point mutation), 
    'insert' (insert a base), 'delete' (delete a base), 'invert'
    (invert a stretch of the chromosome), 'duplicate' (duplicate a
    stretch of the chromosome), 'translocate' (translocate a stretch of
    chromosome to another random position). Default = point.
@param rate: probability of mutation per base above background
    mutation rate. Default = 0.01 (1%). No mutation event will ever 
    happen if (rate + background_mutation) is less than zero.
@param start: starting base on the sequence for mutation.
    Default = 0, start of the genome.
@param end: last base on the sequence for mutation. Default = -1, end of
    the genome.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00038">38</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a id="a8ffcfbae7a2bee146a0fa43e1889e36a"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a8ffcfbae7a2bee146a0fa43e1889e36a">&#9670;&nbsp;</a></span>background_mutation</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">copads.genetic.Chromosome.background_mutation</td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00036">36</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<a id="aa88b94f11b83511072d8f79489c6db84"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aa88b94f11b83511072d8f79489c6db84">&#9670;&nbsp;</a></span>base</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">copads.genetic.Chromosome.base</td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00035">35</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<a id="a03e4ebef1842713f8e051fde835c377e"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a03e4ebef1842713f8e051fde835c377e">&#9670;&nbsp;</a></span>sequence</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">copads.genetic.Chromosome.sequence</td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="../../d2/d43/genetic_8py_source.html#l00034">34</a> of file <a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>copads/<a class="el" href="../../d2/d43/genetic_8py_source.html">genetic.py</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="../../d6/dd1/namespacecopads.html">copads</a></li><li class="navelem"><a class="el" href="../../d1/d47/namespacecopads_1_1genetic.html">genetic</a></li><li class="navelem"><a class="el" href="../../df/d45/classcopads_1_1genetic_1_1_chromosome.html">Chromosome</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="../../doxygen.png" alt="doxygen"/></a> 1.8.14 </li>
  </ul>
</div>
</body>
</html>
