<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.14"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Bactome: rapd_primers Namespace Reference</title>
<link href="../../tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../jquery.js"></script>
<script type="text/javascript" src="../../dynsections.js"></script>
<link href="../../navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../resize.js"></script>
<script type="text/javascript" src="../../navtreedata.js"></script>
<script type="text/javascript" src="../../navtree.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
  $(document).ready(initResizable);
/* @license-end */</script>
<link href="../../search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../search/searchdata.js"></script>
<script type="text/javascript" src="../../search/search.js"></script>
<link href="../../doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Bactome
   </div>
   <div id="projectbrief">Toolbox for Studying Bacteria, Archaea, and Complex Taxons</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.14 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "../../search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="../../menudata.js"></script>
<script type="text/javascript" src="../../menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('../../',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('dc/df6/namespacerapd__primers.html','../../');});
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">rapd_primers Namespace Reference</div>  </div>
</div><!--header-->
<div class="contents">
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a43121077173a6e2eb69d1d689838875e"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#a43121077173a6e2eb69d1d689838875e">fasta_seq</a> (fasta='ATCC8739.fasta')</td></tr>
<tr class="separator:a43121077173a6e2eb69d1d689838875e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1589f5014147ab4ae03b6b6712dabd53"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#a1589f5014147ab4ae03b6b6712dabd53">generate_fragments</a> (seq, tablefile, fragment_size=15)</td></tr>
<tr class="separator:a1589f5014147ab4ae03b6b6712dabd53"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae8a4d9a449f7c405fe3e7ab17169e8c3"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#ae8a4d9a449f7c405fe3e7ab17169e8c3">LCS</a> (seq, test)</td></tr>
<tr class="separator:ae8a4d9a449f7c405fe3e7ab17169e8c3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4465d8a33881459b3474c48501355efe"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#a4465d8a33881459b3474c48501355efe">generate_all_LCS</a> (fragfile, LCSfile, min_primer=7, max_interval=198, min_interval=6)</td></tr>
<tr class="separator:a4465d8a33881459b3474c48501355efe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abbcea3887d753807693080aa1af8babd"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#abbcea3887d753807693080aa1af8babd">get_LCS_by_amplicon</a> (LCSfile, amplicon_size)</td></tr>
<tr class="separator:abbcea3887d753807693080aa1af8babd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acb9f93eea92d5e8579da941be4db1836"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#acb9f93eea92d5e8579da941be4db1836">LCS_uniqueness</a> (LCSfile, amplicon_size)</td></tr>
<tr class="separator:acb9f93eea92d5e8579da941be4db1836"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab1ea250b655d7ebf88cadbfa9e36a42e"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#ab1ea250b655d7ebf88cadbfa9e36a42e">LCS_tabulate</a> (LCSfile, max_interval=198, min_interval=6, fragment_size=15, p='yes')</td></tr>
<tr class="separator:ab1ea250b655d7ebf88cadbfa9e36a42e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aba4307377a81c250bf080ebd3ef61972"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#aba4307377a81c250bf080ebd3ef61972">inverse_LCS</a> (LCSfile, inverseLCSfile, max_interval=198, min_interval=6, fragment_size=15)</td></tr>
<tr class="separator:aba4307377a81c250bf080ebd3ef61972"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34fee403de2bed2ab913cf8a2042fe93"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#a34fee403de2bed2ab913cf8a2042fe93">look_for_primers</a> (inverseLCSfile, num_of_amplicons, min_tm, p='yes')</td></tr>
<tr class="separator:a34fee403de2bed2ab913cf8a2042fe93"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a60f0a5117d5d6ce682173a5a3b3fb983"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../dc/df6/namespacerapd__primers.html#a60f0a5117d5d6ce682173a5a3b3fb983">generate_primers_from_fasta</a> (fasta='ATCC8739.fasta', genome_fragment='genome_fragment.table', max_primer_length=15, min_primer_length=7, max_amplicon_length=3100, min_amplicon_length=300, primer_file='primer.table', inverse_primer_file='invprimer.table')</td></tr>
<tr class="separator:a60f0a5117d5d6ce682173a5a3b3fb983"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><pre class="fragment">Generate potential primers for RAPD/RFLP from reference genomic sequence

Copyright 2009 Maurice Ling &lt;mauriceling@acm.org&gt; on behalf of
Maurice Ling, Chin-How Lee, Jack Si-Hao Oon, Kun-Cheng Lee.

@see: Lee, CH, Lee, KC, Oon, JSH, Ling, MHT. 2010. Bactome, I: 
Python in DNA Fingerprinting. In: Peer-Reviewed Articles from 
PyCon Asia-Pacific 2010. The Python Papers 5(3): 6.

Licence: GNU General Public License version 3

Bactome package is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see &lt;http://www.gnu.org/licenses/&gt;.
</pre> </div><h2 class="groupheader">Function Documentation</h2>
<a id="a43121077173a6e2eb69d1d689838875e"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a43121077173a6e2eb69d1d689838875e">&#9670;&nbsp;</a></span>fasta_seq()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.fasta_seq </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fasta</em> = <code>'ATCC8739.fasta'</code></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Open fasta format file and returns the sequence.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00035">35</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="a4465d8a33881459b3474c48501355efe"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a4465d8a33881459b3474c48501355efe">&#9670;&nbsp;</a></span>generate_all_LCS()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.generate_all_LCS </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fragfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>LCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_primer</em> = <code>7</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>max_interval</em> = <code>198</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_interval</em> = <code>6</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Generates primers where forward primer sequence = reverse primer sequence
given the size of amplicon and minimum length of primers.

Parameters:
    fragfile = file of sliced sequence (tablefile parameter of 
        generate_fragments function)
    LCSfile = name of file to store least common substrings' (primers) data 
    min_primer = smallest acceptable length for primers
    max_interval = number of maximum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 198 as 3000bp amplicon is the maximum normal
        Taq polymerase can amplify. 198 intervals of 15 bases per 
        fragment + 2 flanking fragment of maximum of 15 bases (primer
        length) gives a total of 200 fragments of 15 bases = 3000 bases.
    min_interval = number of minimum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 6 as 120bp amplicon is the minimum length visible
        on 1.5% agarose gel that is also suitable for 3000 bp. 6 intervals 
        of 15 bases per fragment + 2 flanking fragment of maximum of 15 
        bases (primer length) gives a total of 8 fragments of 15 bases = 
        120 bases.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00089">89</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="a1589f5014147ab4ae03b6b6712dabd53"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a1589f5014147ab4ae03b6b6712dabd53">&#9670;&nbsp;</a></span>generate_fragments()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.generate_fragments </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>tablefile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fragment_size</em> = <code>15</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Slice an entire sequence into fragments of equal lengths

Parameters:
    seq = sequence to be sliced (output from fasta_seq function)
    tablefile = name of file to store the sliced sequence
    fragment_size = length of each sequence to be sliced into. This will
        be the maximum primer/probe length.
        Default value is 15 bases.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00042">42</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="a60f0a5117d5d6ce682173a5a3b3fb983"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a60f0a5117d5d6ce682173a5a3b3fb983">&#9670;&nbsp;</a></span>generate_primers_from_fasta()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.generate_primers_from_fasta </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fasta</em> = <code>'ATCC8739.fasta'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>genome_fragment</em> = <code>'genome_fragment.table'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>max_primer_length</em> = <code>15</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_primer_length</em> = <code>7</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>max_amplicon_length</em> = <code>3100</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_amplicon_length</em> = <code>300</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>primer_file</em> = <code>'primer.table'</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>inverse_primer_file</em> = <code>'invprimer.table'</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Generate a file of primers with amplicon size and genomic position from
the Fasta sequence file.

Parameters:
    fasta = name of fasta file to process
    genome_fragment = name of file to store the sliced sequence.
        Default = 'genome_fragment.table'
    max_primer_length = maximum length of primer (fragment size of genome
        slices). Default = 15 bp
    min_primer_length = minimum length of primer. Default = 7 bp
    max_amplicon_length = maximum size of amplicon. Default = 3100 bp
    min_amplicon_length = minimum size of amplicon. Default = 300 bp
    primer_file = file of least common substrings' (primers) data.
        Default = 'primer.table'
    inverse_primer_file = name of primers index file. 
        Default = 'invprimer.table'

Output files:
    1. genome fragment file from 'generate_fragments' function
    2. primer file from 'generate_all_LCS' function
    3. inverse primer file from 'inverse_LCS' function
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00357">357</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="abbcea3887d753807693080aa1af8babd"></a>
<h2 class="memtitle"><span class="permalink"><a href="#abbcea3887d753807693080aa1af8babd">&#9670;&nbsp;</a></span>get_LCS_by_amplicon()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.get_LCS_by_amplicon </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>LCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>amplicon_size</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Extract all the primers and amplicon positions from the LCSfile given the
amplicon length.

Parameters:
    LCSfile = file of least common substrings' (primers) data
    amplicon_size = size of amplified product in terms of number of
        fragments intervening 2 primers. For example, if the fragment 
        size is 15 and the amplicon size is 60, we are then looking
        for primers (forward primer sequence = reverse primer
        sequence) that gives amplicons of 900 bp (60*15) excluding
        the flanking primers.

Returns:
    (List of primers,
    List of amplicons' data)
    where
    1. each data element in "List of amplicons' data" is in the format of
       '&lt;start fragment position&gt;|&lt;end fragment position&gt;|&lt;amplicon size&gt;'
        &lt;start fragment position&gt; and &lt;end fragment position&gt; multiplied 
        by the fragment size will give the corresponding position on the 
        fasta sequence. For example, '89572|89632|60' represents the 
        amplicon from 1343580 bp (889572*15) to 1344480 bp (89632*15) of 
        the fasta sequence, corresponding to an amplicon size of 900bp 
        (60*15).
    2. the number of primers = number of primers's data. This means that in
        primers = get_LCS_by_amplicon('primer.table', 60)
        primers[0][10] sequence will amplify primers[1][10]
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00148">148</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="aba4307377a81c250bf080ebd3ef61972"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aba4307377a81c250bf080ebd3ef61972">&#9670;&nbsp;</a></span>inverse_LCS()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.inverse_LCS </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>LCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>inverseLCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>max_interval</em> = <code>198</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_interval</em> = <code>6</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fragment_size</em> = <code>15</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Generates the index file of LCSfile. 

Format of LCSfile: &lt;start fragment position&gt;|
                   &lt;end fragment position&gt;|
                   &lt;amplicon size&gt; = &lt;primer sequence&gt;
Format of inverseLCSfile: &lt;primer sequence&gt; =
                          [&lt;start fragment position&gt;|
                           &lt;end fragment position&gt;|
                           &lt;amplicon size&gt;, ...]

Parameters:
    LCSfile = file of least common substrings' (primers) data
    inverseLCSfile = name of file to store the index of LCSfile
    max_interval = number of maximum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 198 as 3000bp amplicon is the maximum normal
        Taq polymerase can amplify. 198 intervals of 15 bases per 
        fragment + 2 flanking fragment of maximum of 15 bases (primer
        length) gives a total of 200 fragments of 15 bases = 3000 bases.
    min_interval = number of minimum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 6 as 120bp amplicon is the minimum length visible
        on 1.5% agarose gel that is also suitable for 3000 bp. 6 intervals 
        of 15 bases per fragment + 2 flanking fragment of maximum of 15 
        bases (primer length) gives a total of 8 fragments of 15 bases = 
        120 bases.
    fragment_size = length of each sequence to be sliced into. This will
        be the maximum primer/probe length.
        Default value is 15 bases.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00266">266</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="ae8a4d9a449f7c405fe3e7ab17169e8c3"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae8a4d9a449f7c405fe3e7ab17169e8c3">&#9670;&nbsp;</a></span>LCS()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.LCS </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>test</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Finds the longest common substring between the 2 inputs (seq and test).

Adapted from http://en.wikibooks.org/wiki/Algorithm_Implementation/
Strings/Longest_common_substring
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00064">64</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="ab1ea250b655d7ebf88cadbfa9e36a42e"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab1ea250b655d7ebf88cadbfa9e36a42e">&#9670;&nbsp;</a></span>LCS_tabulate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.LCS_tabulate </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>LCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>max_interval</em> = <code>198</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_interval</em> = <code>6</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fragment_size</em> = <code>15</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>p</em> = <code>'yes'</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Provides a tabulation of the given LCSfile, grouped by amplicon size.

Parameters:
    LCSfile = file of least common substrings' (primers) data
    max_interval = number of maximum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 198 as 3000bp amplicon is the maximum normal
        Taq polymerase can amplify. 198 intervals of 15 bases per 
        fragment + 2 flanking fragment of maximum of 15 bases (primer
        length) gives a total of 200 fragments of 15 bases = 3000 bases.
    min_interval = number of minimum fragment intervals between 2 primers. 
        This value is determines the maximum length of amplicon generated 
        and is determined by the length of each sliced fragment.
        Default value is 6 as 120bp amplicon is the minimum length visible
        on 1.5% agarose gel that is also suitable for 3000 bp. 6 intervals 
        of 15 bases per fragment + 2 flanking fragment of maximum of 15 
        bases (primer length) gives a total of 8 fragments of 15 bases = 
        120 bases.
    fragment_size = length of each sequence to be sliced into. This will
        be the maximum primer/probe length.
        Default value is 15 bases.
    p = flag to determine if the tabulation data is to be printed. 
        Default = yes
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00219">219</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="acb9f93eea92d5e8579da941be4db1836"></a>
<h2 class="memtitle"><span class="permalink"><a href="#acb9f93eea92d5e8579da941be4db1836">&#9670;&nbsp;</a></span>LCS_uniqueness()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.LCS_uniqueness </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>LCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>amplicon_size</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Extract all the primers and amplicon positions from the LCSfile given the
amplicon length and groups the primers into 2 groups: those that only 
yield one amplicon and those that yield more than one amplicons.

Parameters:
    LCSfile = file of least common substrings' (primers) data
    amplicon_size = size of amplified product in terms of number of
        fragments intervening 2 primers. For example, if the fragment 
        size is 15 and the amplicon size is 60, we are then looking
        for primers (forward primer sequence = reverse primer
        sequence) that gives amplicons of 900 bp (60*15) excluding
        the flanking primers.

Returns:
    (List of unique primers,
    List of non-unique primers)
    where 'List of unique primers' are primers amplifying only one 
    amplicon, and 'List of non-unique primers' is a tuple of 'number
    of amplicons' and primer sequence.
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00187">187</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
<a id="a34fee403de2bed2ab913cf8a2042fe93"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a34fee403de2bed2ab913cf8a2042fe93">&#9670;&nbsp;</a></span>look_for_primers()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def rapd_primers.look_for_primers </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>inverseLCSfile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>num_of_amplicons</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>min_tm</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>p</em> = <code>'yes'</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Scans the LCSfile index for primer(s) that amplifies a given number of
amplicons. The primer must not end with adenosine and thymidine, and
must meet minimum annealing temperature.

Parameters:
    inverseLCSfile = name of LCSfile index file
    num_of_amplicons = number of amplicons generated by required primer(s)
    min_tm = minimum annealing temperature (in degrees centigrade) of 
        primers. Calculated as 2AT + 4GC.
    p = flag to determine if data is to be printed. Default = yes

Returns:
    List of primers meeting the criteria
</pre> 
<p class="definition">Definition at line <a class="el" href="../../da/da7/rapd__primers_8py_source.html#l00318">318</a> of file <a class="el" href="../../da/da7/rapd__primers_8py_source.html">rapd_primers.py</a>.</p>

</div>
</div>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="../../dc/df6/namespacerapd__primers.html">rapd_primers</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="../../doxygen.png" alt="doxygen"/></a> 1.8.14 </li>
  </ul>
</div>
</body>
</html>
