<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.14"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Bactome: cub Namespace Reference</title>
<link href="../../tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../jquery.js"></script>
<script type="text/javascript" src="../../dynsections.js"></script>
<link href="../../navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../resize.js"></script>
<script type="text/javascript" src="../../navtreedata.js"></script>
<script type="text/javascript" src="../../navtree.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
  $(document).ready(initResizable);
/* @license-end */</script>
<link href="../../search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="../../search/searchdata.js"></script>
<script type="text/javascript" src="../../search/search.js"></script>
<link href="../../doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Bactome
   </div>
   <div id="projectbrief">Toolbox for Studying Bacteria, Archaea, and Complex Taxons</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.14 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "../../search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="../../menudata.js"></script>
<script type="text/javascript" src="../../menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('../../',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('d8/d1e/namespacecub.html','../../');});
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#nested-classes">Classes</a> &#124;
<a href="#func-members">Functions</a> &#124;
<a href="#var-members">Variables</a>  </div>
  <div class="headertitle">
<div class="title">cub Namespace Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Codon Usage Bias Calculator.  
<a href="#details">More...</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Classes</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">class &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../db/dec/classcub_1_1_codon_usage_bias.html">CodonUsageBias</a></td></tr>
<tr class="memdesc:"><td class="mdescLeft">&#160;</td><td class="mdescRight">Class to hold the core methods for codon usage bias analysis.  <a href="../../db/dec/classcub_1_1_codon_usage_bias.html#details">More...</a><br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a468fa3a2ea2a0c9f2d9b583bce27a28c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#a468fa3a2ea2a0c9f2d9b583bce27a28c">sequenceIDs</a> (fastafile)</td></tr>
<tr class="memdesc:a468fa3a2ea2a0c9f2d9b583bce27a28c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to print out the sequence IDs of all the FASTA records in the FASTA file.  <a href="#a468fa3a2ea2a0c9f2d9b583bce27a28c">More...</a><br /></td></tr>
<tr class="separator:a468fa3a2ea2a0c9f2d9b583bce27a28c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac83e216c91086d232e97f67bd382a33d"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#ac83e216c91086d232e97f67bd382a33d">sequenceDescriptions</a> (fastafile)</td></tr>
<tr class="memdesc:ac83e216c91086d232e97f67bd382a33d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to print out the sequence IDs and descriptions of all the FASTA records in the FASTA file.  <a href="#ac83e216c91086d232e97f67bd382a33d">More...</a><br /></td></tr>
<tr class="separator:ac83e216c91086d232e97f67bd382a33d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac4ccc6329135926565856133550521de"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#ac4ccc6329135926565856133550521de">translate</a> (fastafile, genetic_code=1)</td></tr>
<tr class="memdesc:ac4ccc6329135926565856133550521de"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to translate all the FASTA records from nucleotide sequence(s) to amino acid sequence(s).  <a href="#ac4ccc6329135926565856133550521de">More...</a><br /></td></tr>
<tr class="separator:ac4ccc6329135926565856133550521de"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a13144f4c7243ea5bb36513096b58f1ae"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#a13144f4c7243ea5bb36513096b58f1ae">aminoacidCount</a> (fastafile, genetic_code=1)</td></tr>
<tr class="memdesc:a13144f4c7243ea5bb36513096b58f1ae"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to translate each nucleotide sequence (by FASTA record) and generate a frequency table of the amino acids.  <a href="#a13144f4c7243ea5bb36513096b58f1ae">More...</a><br /></td></tr>
<tr class="separator:a13144f4c7243ea5bb36513096b58f1ae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aff71062122e454e05a44f9de57ab430c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#aff71062122e454e05a44f9de57ab430c">peptideLength</a> (fastafile, genetic_code=1)</td></tr>
<tr class="memdesc:aff71062122e454e05a44f9de57ab430c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to translate each nucleotide sequence (by FASTA record) and count the number of amino acids (peptide length).  <a href="#aff71062122e454e05a44f9de57ab430c">More...</a><br /></td></tr>
<tr class="separator:aff71062122e454e05a44f9de57ab430c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a557fb94d4651c05465070632b4eb8aaa"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#a557fb94d4651c05465070632b4eb8aaa">nucleotideLength</a> (fastafile)</td></tr>
<tr class="memdesc:a557fb94d4651c05465070632b4eb8aaa"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to count the number of nucleotides (number of bases) in each FASTA record.  <a href="#a557fb94d4651c05465070632b4eb8aaa">More...</a><br /></td></tr>
<tr class="separator:a557fb94d4651c05465070632b4eb8aaa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8533cd28eb987635dc13bc243a6d34d2"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#a8533cd28eb987635dc13bc243a6d34d2">flattenCodonCount</a> (CC)</td></tr>
<tr class="memdesc:a8533cd28eb987635dc13bc243a6d34d2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to flatten the codon usage / frequency table by removing the amino acid identifiers.  <a href="#a8533cd28eb987635dc13bc243a6d34d2">More...</a><br /></td></tr>
<tr class="separator:a8533cd28eb987635dc13bc243a6d34d2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae07ced24565f97f101d1a2dec705866c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#ae07ced24565f97f101d1a2dec705866c">codonCount</a> (fastafile, genetic_code=1)</td></tr>
<tr class="memdesc:ae07ced24565f97f101d1a2dec705866c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Function to generate the codon usage frequency table by each FASTA record.  <a href="#ae07ced24565f97f101d1a2dec705866c">More...</a><br /></td></tr>
<tr class="separator:ae07ced24565f97f101d1a2dec705866c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="var-members"></a>
Variables</h2></td></tr>
<tr class="memitem:acbc98a84cb8f7045ea0435662bac8818"><td class="memItemLeft" align="right" valign="top">dictionary&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="../../d8/d1e/namespacecub.html#acbc98a84cb8f7045ea0435662bac8818">exposed_functions</a></td></tr>
<tr class="separator:acbc98a84cb8f7045ea0435662bac8818"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Codon Usage Bias Calculator. </p>
<p>Date created: 3rd May 2018</p>
<p>License: GNU General Public License version 3 for academic or not-for-profit use only</p>
<p>Bactome package is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.</p>
<p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.</p>
<p>You should have received a copy of the GNU General Public License along with this program. If not, see <a href="http://www.gnu.org/licenses/">http://www.gnu.org/licenses/</a>. </p>
</div><h2 class="groupheader">Function Documentation</h2>
<a id="a13144f4c7243ea5bb36513096b58f1ae"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a13144f4c7243ea5bb36513096b58f1ae">&#9670;&nbsp;</a></span>aminoacidCount()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.aminoacidCount </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>genetic_code</em> = <code>1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to translate each nucleotide sequence (by FASTA record) and generate a frequency table of the amino acids. </p>
<p>Usage: </p><pre class="fragment">python cub.py aacount --fastafile=&lt;FASTA file path&gt; 
--genetic_code=&lt;genetic code number&gt;
</pre><p>The output will be in the format of </p><pre class="fragment">&lt;sequence ID&gt; : &lt;A count&gt; : &lt;C count&gt; : &lt;D count&gt; : &lt;E count&gt; 
: &lt;F count&gt; : &lt;G count&gt; : &lt;H count&gt; : &lt;I count&gt; : &lt;K count&gt; : 
&lt;L count&gt; : &lt;M count&gt; : &lt;N count&gt; : &lt;P count&gt; : &lt;Q count&gt; : 
&lt;R count&gt; : &lt;S count&gt; : &lt;T count&gt; : &lt;V count&gt; : &lt;W count&gt; : 
&lt;Y count&gt;
</pre><p>where</p><ul>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>the counts are the number of the respective amino acid; for example, A count is the number of alanine in the peptide</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
    <tr><td class="paramname">genetic_code</td><td>Integer: Genetic code number to be used for translation. Default = 1 (Standard Code). For more information, see <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a> </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00187">187</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="ae07ced24565f97f101d1a2dec705866c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae07ced24565f97f101d1a2dec705866c">&#9670;&nbsp;</a></span>codonCount()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.codonCount </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>genetic_code</em> = <code>1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to generate the codon usage frequency table by each FASTA record. </p>
<p>Usage: </p><pre class="fragment">python cub.py codoncount --fastafile=&lt;FASTA file path&gt; 
--genetic_code=&lt;genetic code number&gt;
</pre><p>The output will be in the format of </p><pre class="fragment">&lt;sequence ID&gt; : &lt;AAA count&gt; : &lt;AAC count&gt; : &lt;AAG count&gt; : 
&lt;AAT count&gt; : &lt;ACA count&gt; : &lt;ACC count&gt; : &lt;ACG count&gt; : 
&lt;ACT count&gt; : &lt;AGA count&gt; : &lt;AGC count&gt; : &lt;AGG count&gt; : 
&lt;AGT count&gt; : &lt;ATA count&gt; : &lt;ATC count&gt; : &lt;ATG count&gt; : 
&lt;ATT count&gt; : &lt;CAA count&gt; : &lt;CAC count&gt; : &lt;CAG count&gt; : 
&lt;CAT count&gt; : &lt;CCA count&gt; : &lt;CCC count&gt; : &lt;CCG count&gt; : 
&lt;CCT count&gt; : &lt;CGA count&gt; : &lt;CGC count&gt; : &lt;CGG count&gt; : 
&lt;CGT count&gt; : &lt;CTA count&gt; : &lt;CTC count&gt; : &lt;CTG count&gt; : 
&lt;CTT count&gt; : &lt;GAA count&gt; : &lt;GAC count&gt; : &lt;GAG count&gt; : 
&lt;GAT count&gt; : &lt;GCA count&gt; : &lt;GCC count&gt; : &lt;GCG count&gt; : 
&lt;GCT count&gt; : &lt;GGA count&gt; : &lt;GGC count&gt; : &lt;GGG count&gt; : 
&lt;GGT count&gt; : &lt;GTA count&gt; : &lt;GTC count&gt; : &lt;GTG count&gt; : 
&lt;GTT count&gt; : &lt;TAA count&gt; : &lt;TAC count&gt; : &lt;TAG count&gt; : 
&lt;TAT count&gt; : &lt;TCA count&gt; : &lt;TCC count&gt; : &lt;TCG count&gt; : 
&lt;TCT count&gt; : &lt;TGA count&gt; : &lt;TGC count&gt; : &lt;TGG count&gt; : 
&lt;TGT count&gt; : &lt;TTA count&gt; : &lt;TTC count&gt; : &lt;TTG count&gt; : 
&lt;TTT count&gt;
</pre><p>where</p><ul>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>the counts are the number of each codon; for example, AAA count is the number of AAA codon in the sequence</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
    <tr><td class="paramname">genetic_code</td><td>Integer: Genetic code number to be used for translation. Default = 1 (Standard Code). For more information, see <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a> </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00316">316</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="a8533cd28eb987635dc13bc243a6d34d2"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a8533cd28eb987635dc13bc243a6d34d2">&#9670;&nbsp;</a></span>flattenCodonCount()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.flattenCodonCount </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>CC</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to flatten the codon usage / frequency table by removing the amino acid identifiers. </p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">CC</td><td>Codon usage table as a nested dictionary of {'amino acid': {'codon': count}}.</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Returns</dt><dd>Dictionary of {'codon': count}. </dd></dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00285">285</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="a557fb94d4651c05465070632b4eb8aaa"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a557fb94d4651c05465070632b4eb8aaa">&#9670;&nbsp;</a></span>nucleotideLength()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.nucleotideLength </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to count the number of nucleotides (number of bases) in each FASTA record. </p>
<p>Usage: </p><pre class="fragment">python cub.py nlength --fastafile=&lt;FASTA file path&gt; 
</pre><p>The output will be in the format of</p>
<p>&lt;sequence id&gt;=""&gt; : &lt;nucleotide length&gt;=""&gt;</p>
<p>where</p><ul>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>nucleotide length is the number of bases in the peptide</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00260">260</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="aff71062122e454e05a44f9de57ab430c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#aff71062122e454e05a44f9de57ab430c">&#9670;&nbsp;</a></span>peptideLength()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.peptideLength </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>genetic_code</em> = <code>1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to translate each nucleotide sequence (by FASTA record) and count the number of amino acids (peptide length). </p>
<p>Usage: </p><pre class="fragment">python cub.py plength --fastafile=&lt;FASTA file path&gt; 
--genetic_code=&lt;genetic code number&gt;
</pre><p>The output will be in the format of</p>
<p>&lt;sequence id&gt;=""&gt; : &lt;peptide length&gt;=""&gt;</p>
<p>where</p><ul>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>peptide length is the number of amino acids in the peptide</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
    <tr><td class="paramname">genetic_code</td><td>Integer: Genetic code number to be used for translation. Default = 1 (Standard Code). For more information, see <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a> </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00231">231</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="ac83e216c91086d232e97f67bd382a33d"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ac83e216c91086d232e97f67bd382a33d">&#9670;&nbsp;</a></span>sequenceDescriptions()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.sequenceDescriptions </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to print out the sequence IDs and descriptions of all the FASTA records in the FASTA file. </p>
<p>Usage: </p><pre class="fragment">python cub.py showDesc --fastafile=&lt;FASTA file path&gt;
</pre><p>The output will be in the format of </p><pre class="fragment">&lt;count&gt; : &lt;sequence ID&gt; : &lt;description&gt;
</pre><p>where</p><ul>
<li>count is the numeric running order</li>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>description is the description of the FASTA record</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00130">130</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="a468fa3a2ea2a0c9f2d9b583bce27a28c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a468fa3a2ea2a0c9f2d9b583bce27a28c">&#9670;&nbsp;</a></span>sequenceIDs()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.sequenceIDs </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to print out the sequence IDs of all the FASTA records in the FASTA file. </p>
<p>Usage: </p><pre class="fragment">python cub.py showIDs --fastafile=&lt;FASTA file path&gt;
</pre><p>The output will be in the format of </p><pre class="fragment">&lt;count&gt; : &lt;sequence ID&gt;
</pre><p>where</p><ul>
<li>count is the numeric running order</li>
<li>sequence ID is the sequence ID of the FASTA record.</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00104">104</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<a id="ac4ccc6329135926565856133550521de"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ac4ccc6329135926565856133550521de">&#9670;&nbsp;</a></span>translate()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def cub.translate </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>fastafile</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>genetic_code</em> = <code>1</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Function to translate all the FASTA records from nucleotide sequence(s) to amino acid sequence(s). </p>
<p>Usage: </p><pre class="fragment">python cub.py translate --fastafile=&lt;FASTA file path&gt; 
--genetic_code=&lt;genetic code number&gt;
</pre><p>The output will be in the format of </p><pre class="fragment">&lt;sequence ID&gt; : &lt;amino acid sequence&gt;
</pre><p>where</p><ul>
<li>sequence ID is the sequence ID of the FASTA record</li>
<li>amino acid sequence is the translated amino acid sequence of the FASTA record</li>
</ul>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">fastafile</td><td>String: Path to the FASTA file to be processed. </td></tr>
    <tr><td class="paramname">genetic_code</td><td>Integer: Genetic code number to be used for translation. Default = 1 (Standard Code). For more information, see <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a> </td></tr>
  </table>
  </dd>
</dl>

<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00157">157</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
<h2 class="groupheader">Variable Documentation</h2>
<a id="acbc98a84cb8f7045ea0435662bac8818"></a>
<h2 class="memtitle"><span class="permalink"><a href="#acbc98a84cb8f7045ea0435662bac8818">&#9670;&nbsp;</a></span>exposed_functions</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">dictionary cub.exposed_functions</td>
        </tr>
      </table>
</div><div class="memdoc">
<b>Initial value:</b><div class="fragment"><div class="line"><a name="l00001"></a><span class="lineno">    1</span>&#160;=  {<span class="stringliteral">&#39;showIDs&#39;</span>: sequenceIDs,</div><div class="line"><a name="l00002"></a><span class="lineno">    2</span>&#160;                         <span class="stringliteral">&#39;showDesc&#39;</span>: sequenceDescriptions,</div><div class="line"><a name="l00003"></a><span class="lineno">    3</span>&#160;                         <span class="stringliteral">&#39;plength&#39;</span>: peptideLength,</div><div class="line"><a name="l00004"></a><span class="lineno">    4</span>&#160;                         <span class="stringliteral">&#39;nlength&#39;</span>: nucleotideLength,</div><div class="line"><a name="l00005"></a><span class="lineno">    5</span>&#160;                         <span class="stringliteral">&#39;translate&#39;</span>: translate,</div><div class="line"><a name="l00006"></a><span class="lineno">    6</span>&#160;                         <span class="stringliteral">&#39;codoncount&#39;</span>: codonCount,</div><div class="line"><a name="l00007"></a><span class="lineno">    7</span>&#160;                         <span class="stringliteral">&#39;aacount&#39;</span>: aminoacidCount}</div></div><!-- fragment -->
<p class="definition">Definition at line <a class="el" href="../../d2/d28/cub_8py_source.html#l00400">400</a> of file <a class="el" href="../../d2/d28/cub_8py_source.html">cub.py</a>.</p>

</div>
</div>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="../../d8/d1e/namespacecub.html">cub</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="../../doxygen.png" alt="doxygen"/></a> 1.8.14 </li>
  </ul>
</div>
</body>
</html>
